netsock.c

```diff
@@ -209,7 +209,7 @@ ATTR_NONNULL_ALL int netsock_open_dhcp(ddhcp_epoll_data* data,uint16_t port) {
   memcpy(&sin.sin_addr, &address_client, sizeof(sin.sin_addr));
 
   if (setsockopt(sock, SOL_SOCKET, SO_BINDTODEVICE, data->interface_name,
-                 (socklen_t)strlen(data->interface_name) + 1)) {
+                 (socklen_t)strlen(data->interface_name))) {
```

Comment:
I like this, good catch. This could even be an strnlen with maxlen IFNAMSIZ. For even more defensive programming.